### ISSUE TITLE
Quasar Repository Re-organization.

### ISSUE THREAD

**Issue by zeeiyerWork** [precog/quasar#1981]
Currently Quasar is considered to be too monolithic. This results in long build times (e.g. 100 minutes build time , 600 minutes total time). This ticket is the EPIC which will in turn point to multiple smaller tickets. 

The hope is to break out Connectors and other "logical" pieces into their own repositories and then building inter-repository dependencies, so as to achieve the desired level of modularity. This is delicate surgery and will likely entail a "binary-switch" release - i.e. will be too hard to maintain 2 code bases concurrently, so we will just switch from the current monolithic 
Quasar repository to a modular organization of repositories on a certain date/time and then go from there.

**Comment by drostron** [comment]
Coverage has only been running on master since the introduction of PR caching due to how that is achieved. This means that passing PRs can be merged and subsequently cause master to start failing due to a coverage threshold not being met. Currently we're wading the waters and making adjustments as necessary. We should shore this up and start running coverage against PRs again once a poly repo is in place. PR caching shouldn't be necessary afterwards due to smaller and hence faster builds.

**Comment by djspiewak** [comment]
Polyrepoing out the connectors should be relatively straightforward once we fully invert the dependency between `it` and the connectors.  That task requires some restructuring effort, but not too much since most of that work is already done (the current state of things is a very hacky shim, since the connector architecture is now very clearly biased towards an inversion).

Once the dependency is inverted, polyrepo for the connectors becomes a relatively trivial application of git witchcraft (to avoid losing commit history) and infrastructure setup (to ensure we still have Travis stuff).  It also requires moving to a four-component version for Quasar itself, which is a relatively small change, though one which touches some other stuff (including SDBE).
